File: ads/src/main/java/com/ads/module/config/ERainAdConfig.java
package com.ads.module.config;

import android.app.Application;

import java.util.ArrayList;
import java.util.List;

public class ERainAdConfig {

    public static final String ENVIRONMENT_DEVELOP = "develop";
    public static final String ENVIRONMENT_PRODUCTION = "production";

    public static final String DEFAULT_TOKEN_FACEBOOK_SDK = "client_token";

    private boolean isVariantDev = false;

    /**
     * adjustConfig enable adjust and setup adjust token
     */
    private AdjustConfig adjustConfig;

    private String idAdResume;
    private List<String> listDeviceTest = new ArrayList();

    private Application application;
    private boolean enableAdResume = false;
    private String facebookClientToken = DEFAULT_TOKEN_FACEBOOK_SDK;

    /**
     * intervalInterstitialAd: time between two interstitial ad impressions
     * unit: seconds
     */
    private int intervalInterstitialAd = 0;

    public ERainAdConfig(Application application) {
        this.application = application;
    }

    public ERainAdConfig(Application application, String environment) {
        this.isVariantDev = environment.equals(ENVIRONMENT_DEVELOP);
        this.application = application;
    }

    public void setEnvironment(String environment) {
        this.isVariantDev = environment.equals(ENVIRONMENT_DEVELOP);
    }

    public AdjustConfig getAdjustConfig() {
        return adjustConfig;
    }

    public void setAdjustConfig(AdjustConfig adjustConfig) {
        this.adjustConfig = adjustConfig;
    }

    public Application getApplication() {
        return application;
    }


    public Boolean isVariantDev() {
        return isVariantDev;
    }


    public String getIdAdResume() {
        return idAdResume;
    }

    public List<String> getListDeviceTest() {
        return listDeviceTest;
    }

    public void setListDeviceTest(List<String> listDeviceTest) {
        this.listDeviceTest = listDeviceTest;
    }


    public void setIdAdResume(String idAdResume) {
        this.idAdResume = idAdResume;
        enableAdResume = true;
    }

    public Boolean isEnableAdResume() {
        return enableAdResume;
    }


    public Boolean isEnableAdjust() {
        if (adjustConfig == null)
            return false;
        return adjustConfig.isEnableAdjust();
    }

    public int getIntervalInterstitialAd() {
        return intervalInterstitialAd;
    }

    public void setIntervalInterstitialAd(int intervalInterstitialAd) {
        this.intervalInterstitialAd = intervalInterstitialAd;
    }

    public void setFacebookClientToken(String token) {
        this.facebookClientToken = token;
    }

    public String getFacebookClientToken() {
        return this.facebookClientToken;
    }

}

File: ads/src/main/java/com/ads/module/config/ERainAdConfig.kt
package com.ads.module.config

import android.app.Application
import com.ads.module.config.settings.AdBehavior

open class ERainAdConfig(application: Application?) {

    companion object {
        const val ENVIRONMENT_DEVELOP = "develop"
        const val ENVIRONMENT_PRODUCTION = "production"

        const val DEFAULT_TOKEN_FACEBOOK_SDK = "client_token"
    }

    private var variantDev = false

    private var enableAdResume = false

    constructor(application: Application?, environment: String?) : this(application) {
        this.variantDev = environment!! == ENVIRONMENT_DEVELOP
    }

    open fun setEnvironment(environment: String?) {
        this.variantDev = environment!! == ENVIRONMENT_DEVELOP
    }

    private var adjustConfigField: AdjustConfig? = null

    /**
     * adjustConfig enable adjust and setup adjust token
     */
    open var adjustConfig: AdjustConfig?
        get() = adjustConfigField
        set(value) {
            adjustConfigField = value
        }

    open val application: Application? = application

    open val isVariantDev: Boolean?
        get() = variantDev

    /**
     * Sets the app-resume ad unit and switches app-resume on.
     *
     * A blank id switches it back off: enabling it anyway made AppOpenManager request an ad with
     * an empty unit, which GMA rejects with "Cannot determine request type" on every cold start.
     */
    open var idAdResume: String? = null
        set(value) {
            field = value
            enableAdResume = value != null && value.trim { it <= ' ' }.isNotEmpty()
        }

    open var listDeviceTest: List<String>? = ArrayList()

    open val isEnableAdResume: Boolean?
        get() = enableAdResume

    open val isEnableAdjust: Boolean?
        get() = adjustConfigField?.isEnableAdjust ?: false

    /**
     * intervalInterstitialAd: time between two interstitial ad impressions
     * unit: seconds
     */
    open var intervalInterstitialAd: Int =
        (AdBehavior.defaultNumber("interstitial.frequency.interval_ms") / 1000).toInt()

    open var facebookClientToken: String? = DEFAULT_TOKEN_FACEBOOK_SDK
}
